### ISSUE TITLE
Enable Edge Loops and Edge Rings with "Select Circle"

### ISSUE THREAD

**Issue by xan2622** [Stromberg90/Scripts#3]
I use **Context Select** on Blender 2.81 alpha (51be0a765e44-win64) and everything works fine with the Box-Selection but not really with the Circle-Selection, hence this request:

I mostly use the Circle Selection (radius=10) because it allows to select many edges very quickly but unfortunately, when I double click on an edge to create an "Edge Loop" selection, it doesn't work, same for Edge Rings with the "Select Circle" Tool.

It would be awesome if you could make this script compatible with the "Select Circle" Tool.

**Comment by Stromberg90** [comment]
I belive this is a bug in blender, I've reported it here https://developer.blender.org/T68030

**Comment by MightyBOBcnc** [comment]
**TL;DR:** 
The Circle Select tool and the Select operator are both using LMB with a "Mouse" event type in the keymap which produces a conflict.  Change all the entries for Circle Select in the 3D View section from "Mouse" to "Tweak" in the keymap editor and you'll be able to use the Context Select addon while using the Circle Select tool.  (Or change Circle Select to some other button like MMB and then continue using LMB double click to select loops and rings with the addon.)

**Extended answer:** 
The Circle Select tool doesn't actually do single click selection.  It _appears_ to do single click selection if you don't move the mouse across multiple components while using it but actually it _only_ knows how to 'paint' a selection.  Try doing this at the corner (vertex) in edge mode where more than one edge meets; it will select all the edges.  Because, again, it isn't a 'real' click selection, it is 'painting' anything in the cursor.  The Box and Lasso tools are the same way; they don't actually do single click selection and they don't set an active edge.

The difference is that Box and Lasso use "Tweak" events in the keymap which doesn't conflict with the regular Select operator (view3d.select) which uses the "Mouse" event type.  The regular Select operator is what sets an active component.  That's why you can be in the Box Select tool and do a single click on a component using the same button (LMB) that you would use to drag a selection and it becomes selected and active.  It's the Select operator doing it, not the Box Select tool.

Circle Select, however, is using the "Mouse" event, too,  which conflicts with the regular Select operator because that means they're both using LMB and the "Mouse" event type.  So the result is that you get the Circle Tool's painted selection instead of view3d.select because of the order of entries in the keymap; Circle Select "wins" the conflict.

As a test (without changing the Circle Select's event type to "Tweak") you can see how there is no conflict if you change Circle Select to a different button like MMB instead of LMB, and then use LMB to do a single click select while in the Circle Select tool.  Or if you switch to the Blender 27X keymap, for example, you can right click on an edge while in the Circle Select tool and it will become active because RMB is used for view3d.select in that keymap while Circle Select uses LMB to paint a selection; no conflict.
